src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,507 @@
+                                             coordStrings[iFace]);
+        if (!converter->exportToFile(image, filename)) {
+          return false;
+        }
+      } break;
+
+      case TextureType::Depth: {
+        filename = Cr::Utility::formatString("{}.{}.{}.hdr", imageFilePrefix,
+                                             getTextureTypeFilenameString(type),
+                                             coordStrings[iFace]);
+        Mn::Image2D image = textures_[type]->image(
+            convertFaceIndexToCubeMapCoordinate(iFace), 0,
+            {Mn::GL::PixelFormat::DepthComponent, Mn::GL::PixelType::Float});
+        Mn::ImageView2D depthAsRedChannel{
+            image.storage(), Mn::PixelFormat::R32F, image.size(), image.data()};
+        if (!converter->exportToFile(depthAsRedChannel, filename)) {
+          return false;
+        }
+        // XXX
+        // display 8 floats
+        /*
+        Mn::Debug{} << "========"
+                    << Cr::Containers::arrayCast<float>(
+                           image.data().prefix(32));
+        Mn::Debug{} << "image size in bytes: " << image.data().size();
+        */
+      } break;
+    }
+    CORRADE_ASSERT(!filename.empty(),
+                   "CubeMap::saveTexture(): Unknown texture type.", false);
+
+    LOG(INFO) << "Saved image " << iFace << " to " << filename;
+  }  // for
+
+  return true;
+}
+#endif
+
+void CubeMap::loadTexture(TextureType type,
+                          const std::string& imageFilePrefix,
+                          const std::string& imageFileExtension) {
+  textureTypeSanityCheck(flags_, type, "CubeMap::loadTexture():");
+
+  // plugin manager used to instantiate importers which in turn are used
+  // to load image data
+  Cr::PluginManager::Manager<Mn::Trade::AbstractImporter> manager;
+  Cr::Containers::Pointer<Mn::Trade::AbstractImporter> importer =
+      manager.loadAndInstantiate("AnyImageImporter");
+  CORRADE_INTERNAL_ASSERT(importer);
+
+  const char* coordStrings[6] = {"+X", "-X", "+Y", "-Y", "+Z", "-Z"};
+  int imageSize = 0;
+
+  // set images
+  Mn::GL::CubeMapTexture* texture = nullptr;
+  switch (type) {
+    case TextureType::Color:
+      texture = textures_[TextureType::Color].get();
+      break;
+
+    case TextureType::Depth:
+      texture = textures_[TextureType::Depth].get();
+      break;
+  }
+  CORRADE_ASSERT(texture, "CubeMap::loadTexture(): Unknown texture type.", );
+
+  for (int iFace = 0; iFace < 6; ++iFace) {
+    // open image file
+
+    std::string filename = Cr::Utility::formatString(
+        "{}.{}.{}.{}", imageFilePrefix, getTextureTypeFilenameString(type),
+        coordStrings[iFace], imageFileExtension);
+
+    importer->openFile(filename);
+    Cr::Containers::Optional<Mn::Trade::ImageData2D> imageData =
+        importer->image2D(0);
+
+    // XXX
+    /*
+    Mn::Debug{} << "imagedata raw data size = " << (*imageData).data().size();
```

Comment:
I printed out the size, which was 17280000, and it is correct. because 17280000 = 5760000 x 3.
3 means when the image (R32F) is saved to .hdr format, it is expended to RGB32F. And 5760000 matched with the number printed out in the `saveTexture(...)`. See my previous comments in that function.